dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/cache/AnalyticsCacheSettings.java

```diff
@@ -121,25 +128,9 @@ public long fixedExpirationTimeOrDefault()
         final CacheStrategy cacheStrategy = systemSettingManager.getSystemSetting( CACHE_STRATEGY,
             CacheStrategy.class );
 
-        if ( cacheStrategy != null && cacheStrategy.hasExpirationTimeSet() )
-        {
-            return cacheStrategy.toSeconds();
-        }
-        else
-        {
-            // Try to get a default value
-            final CacheStrategy defaultExpirationTime = (CacheStrategy) CACHE_STRATEGY.getDefaultValue();
-
-            if ( defaultExpirationTime.hasExpirationTimeSet() )
-            {
-                return defaultExpirationTime.toSeconds();
-            }
-            else
-            {
-                // Return ZERO (always expire)
-                return NO_CACHE.toSeconds();
-            }
-        }
+        return (NO_CACHE.equals( cacheStrategy ))
```

Comment:
Hi @jimgrace,

With this change, it seems that we can only return the expiration time of `0` or `2 weeks`. It means that the user can only choose between `2 weeks` and `0`.
In this case, do we want to ignore the other possible caching timeouts defined by the users, defined in `CacheStrategy`?

If this is the objective it is alright, but then I think it would be better to remove all other possible timeouts in `CacheStrategy` and also remove them from the UI...

Reply:
@maikelarabori Thanks for the question. The idea is that the `CacheStrategy` will still determine the setting in the browser Cache-Control requests that come from the user, but effectively the backend data cache will be either NONE or 2 weeks. Allowing the user to select the `CacheStrategy` will still give them the choice of how the Cache-Control will be set in the browser, and hence how stale the data may appear from that perspective. But in the backend cache the data will never be stale because it is being flushed on analytics rebuild.

We will have to document this carefully.

In Slack messaging including PEPFAR, Lars, Phil, and me, we agreed that flushing the data cache in the backend on analytics rebuild was probably the right thing to do. I wrote this PR to see if this was a reasonable approach to implementing this. It looks like Lars wants your opinion on this too.

Under the circumstances, I think it might be reasonable to just delete all the `PROGRESSIVE` cache logic from the backend, and remove that option from the frontend as well. But it looked like it was a lot of work to implement that option in the first place, and I assumed it was done for good reason to support some installations. So at least for the moment I left it in. (It also means that we don't have to change the frontend at all for now.) But I could be persuaded that we should remove it now and change the frontend as well.

Reply:
Thanks for the great clarification @jimgrace. That makes sense :+1: 
Regarding the `PROGRESSIVE` thing, that was a requirement I was asked to implement some time ago.
I agree that it introduces extra complexity. If for some reason this can be dropped or is not really used by clients, I would certainly agree to remove it in order to keep things simple. But as you well said, some installations might miss it. This would be the only concern I guess.

Cheers,
Maikel

Reply:
Good points. Right I think we probably can remove the progressive caching. I certainly know this was my idea at the time and the thinking was that older data could have longer cache TTL than newer data. Though with the analytics tables and cache being in sync, this may perhaps be outdated now.